spanner_dbapi/cursor.py

```diff
@@ -77,39 +80,38 @@ def execute(self, sql, args=None):
 
             # For every other operation, we've got to ensure that
             # any prior DDL statements were run.
-            self._run_prior_DDL_statements()
+            self._run_prior_ddl_statements()
 
             if classification == STMT_NON_UPDATING:
-                self.__handle_DQL(sql, args or None)
+                self._handle_dql(sql, args)
             elif classification == STMT_INSERT:
-                self.__handle_insert(sql, args or None)
+                self._handle_insert(sql, args)
             else:
-                self.__handle_update(sql, args or None)
+                self._handle_update(sql, args)
         except (grpc_exceptions.AlreadyExists, grpc_exceptions.FailedPrecondition) as e:
-            raise IntegrityError(e.details if hasattr(e, 'details') else e)
+            raise IntegrityError(e.details if hasattr(e, "details") else e)
         except grpc_exceptions.InvalidArgument as e:
-            raise ProgrammingError(e.details if hasattr(e, 'details') else e)
+            raise ProgrammingError(e.details if hasattr(e, "details") else e)
         except grpc_exceptions.InternalServerError as e:
-            raise OperationalError(e.details if hasattr(e, 'details') else e)
+            raise OperationalError(e.details if hasattr(e, "details") else e)
 
-    def __handle_update(self, sql, params):
-        self._connection.in_transaction(
-            self.__do_execute_update,
-            sql, params,
-        )
+    def _handle_update(self, sql, params):
+        ensure_where_clause(sql)
+        self._connection.in_transaction(self.__do_execute_update, sql, params)
 
     def __do_execute_update(self, transaction, sql, params, param_types=None):
-        sql = ensure_where_clause(sql)
```

Comment:
I also realized that it's not very okay to do the check here, as `__do_execute_update()` will be sent to `Database.run_in_transaction()`, and then to `Session.run_in_transaction()` as `func` arg. This last method will start a transaction (if needed) and only then it'll run `__do_execute_update()` (and the check). We better run the check before starting anything. The check call moved to line 99